add 6 system variables introduced in 6.3 but have no user doc (#10564)
pingcap/docs: system-variables.md

- Range: `[0, 9223372036854775807]`
- This variable is used to set the number of retries when the DDL operation fails. When the number of retries exceeds the parameter value, the wrong DDL operation is canceled.

> 

Author: Added in https://github.com/pingcap/docs/pull/9886.

### `tidb_ddl_flashback_concurrency` <span class="version-mark">New in v6.3.0</span>

> **Warning:**
>

> > The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.

Reviewer: ```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

Reviewer: "not recommended" or "do not",  please check it with R&D.

Author: `Do not` is better.

- This variable is used to set whether to enable the statistics `Fast Analyze` feature.
- If the statistics `Fast Analyze` feature is enabled, TiDB randomly samples about 10,000 rows of data as statistics. When the data is distributed unevenly or the data size is small, the statistics accuracy is low. This might lead to a non-optimal execution plan, for example, selecting a wrong index. If the execution time of the regular `Analyze` statement is acceptable, it is recommended to disable the `Fast Analyze` feature.

### `tidb_enable_foreign_key` <span class="version-mark">New in v6.3.0</span>

> **Warning:**
>

> > The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.

Author: ```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

- Scope: GLOBAL
- Persists to cluster: Yes
- Default value: `OFF`
- This variable controls whether to enable the `FOREIGN KEY` feature.

### `tidb_enable_general_plan_cache` <span class="version-mark">New in v6.3.0</span>

> **Warning:**
>

> > The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.

Author: ```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

</CustomContent>

### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">New in v6.3.0</span>

> **Warning:**
>

> > The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.

Author: ```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Default value: `OFF`

> - This variable controls whether read requests in SQL write statements are pushed down to TiFlash.

Reviewer: ```suggestion
- This variable controls whether read requests in SQL write statements can be pushed down to TiFlash.
```

</CustomContent>

### `tidb_general_plan_cache_size` <span class="version-mark">New in v6.3.0</span>

> **Warning:**
>

> > The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.

Author: ```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Default value: `100`
- Range: `[1, 100000]`

> - This variable controls the maximum number of execution plans that can be cached using General Plan Cache.

Reviewer: ```suggestion
- This variable controls the maximum number of execution plans that can be cached by General Plan Cache.
```
